src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java

```diff
@@ -77,49 +78,73 @@ protected <T> List<T[]> getChunks(
         return chunks;
     }
 
-    protected void sendBlockHeadersChunks(
-        List<String> blockHeaders,
-        String actualMethod,
-        boolean keepPreviousChunkLastItem
-    ) throws HSMClientException {
+    private void validateBlockHeaders(List<String> blockHeaders, String methodName) throws HSMClientException {
         if (isStopped) {
-            return;
+            throw new HSMBlockchainBookkeepingRelatedException("[{}] HSM has been stopped." + methodName);
         }
         if (blockHeaders == null || blockHeaders.isEmpty()) {
-            return;
+            throw new HSMBlockchainBookkeepingRelatedException("[{}] Block headers is null or empty." + methodName);
         }
         // If HSM has an advanceBlockchain or updateAncestorBlock in progress, then it can't be called.
         if (getHSMPointer().isInProgress()) {
-            logger.trace("[{}] HSM is already updating its state. Not going to proceed with this request", actualMethod);
-            return;
+            throw new HSMBlockchainBookkeepingRelatedException("[{}] HSM is already updating its state. Not going to proceed with this request." + methodName);
+        }
+    }
+
+    private void addBlocksToPayload(ObjectNode payload, String[] blockHeaderChunk) {
+        final String BLOCKS_FIELD = "blocks";
+        ArrayNode blocksFieldData = new ObjectMapper().createArrayNode();
+        for (String blockHeader : blockHeaderChunk) {
+            blocksFieldData.add(blockHeader);
+        }
+        payload.set(BLOCKS_FIELD, blocksFieldData);
+    }
+
+    private void addBrothersToPayload(ObjectNode payload, List<String[]> brothers) {
+        final String BROTHERS_FIELD = "brothers";
+        ArrayNode brothersFieldData = new ObjectMapper().createArrayNode();
+        for (String[] brotherArray : brothers) {
```

Comment:
```suggestion
        for (String[] blockHeaderBrothers : brothers) {
```
Remember not to use the dat type as the name of variables

Reply:
Updated